cpp11/Ice/interceptor/AuthenticatorI.cpp

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#include <chrono>
+#include <iostream>
+#include <Ice/Ice.h>
+#include <AuthenticatorI.h>
+
+using namespace std;
+
+long getCurrentTimeMillis()
+{
+    auto duration = chrono::system_clock::now().time_since_epoch();
+    return chrono::duration_cast<chrono::milliseconds>(duration).count();
+}
+
+AuthenticatorI::AuthenticatorI() :
+    _rand(getCurrentTimeMillis())
```

Comment:
I think it is preferable to use a random_device to seed the random number generator

``` 
 std::random_device rd;
  _rand(rd()); 
```